bots/reads.ts

```diff
@@ -434,18 +437,31 @@ const handleUpdateUrl = async (url: string, ctx: ReadsContext, config: ReadsConf
 
   if (!cleanUrl.includes('x.com')) {
     // not twitter, so save the title
-    ctx.session.item.title = title || '';
+    ctx.session.item.title = title || "";
+    // Generate and set the summary
+    try {
+      const openai = new OpenAI({
+        apiKey: config.openaiKey,
+      });
+      const summary = await summarizeURL(url,openai || null);
+      ctx.session.item.description =
+        summary.length > 500 ? summary.substring(0, 497) + "..." : summary;
+    } catch (e) {
+      console.error("Error generating summary: ", e);
+      ctx.session.item.description = description || "";
+      await ctx.reply('sorry, generating a summary failed for that url. Please try again');
```

Comment:
let's adjust this since the user doesn't _have_ to try again if they're ok with using the metadata description.

```suggestion
      await ctx.reply('sorry, generating the AI summary failed for that url. falling back to metadata description.');
```

Reply:
Added!